stl/inc/deque

```diff
@@ -1236,52 +1314,84 @@ private:
         }
     };
 
-public:
-    template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
-    iterator insert(const_iterator _Where, _Iter _First, _Iter _Last) {
-        // insert [_First, _Last) at _Where, input iterators
-        size_type _Off = static_cast<size_type>(_Where - begin());
+    enum class _Is_bidi : bool { _No, _Yes };
+
+    template <_Is_bidi _Bidi, class _Iter, class _Sent>
+    iterator _Insert_range(const size_type _Off, _Iter _First, _Sent _Last) {
+        // insert [_First, _Last) at begin() + _Off

```

Comment:
This should definitely document the requirement that `_Bidi == _Is_bidi::_Yes \implies bidirectional_iterator<_Sent>`